lib/response/headers.ex

```diff
@@ -9,6 +13,49 @@ defmodule HTTPServer.Response.Headers do
           allow: list(String.t())
         }
 
+  def new() do
+    %__MODULE__{}
+  end
+
+  def build(res, req = %Request{method: "OPTIONS"}, media_type) do
+    headers =
+      res.headers
+      |> content(media_type, res.body)
+      |> host(req.headers)
+      |> allow(@routes.routes[req.path][:methods])
+
+    %{res | headers: headers}
+  end
```

Comment:
In Elixir it is more common to pattern match off the `def` line, I notice you do not default to the best practice often.

example:
```suggestion
  def build(res = %Response{headers: headers, body: body}, req = %Request{method: "OPTIONS", path: path}, media_type) do
    option_headers =
     headers
      |> content(media_type, body)
      |> host(headers)
      |> allow(@routes.routes[path][:methods])

    %{res | headers: option_headers}
  end
```

Reply:
Having build take both a `%Response` and a `%Request` seems a bit much. I would expect `build` to take a `%Request` and "build" a `%Response`. Is there any way to simplify this?

Reply:
I don't disagree here, but I think the header building functionality is where I keep getting sort of stuck. I had previously created a `default_headers` function in Headers that called content() and host(), then used conditional statements in `build/4` to add additional headers like the pattern matching is doing above. But Response felt like the wrong place for the different Header variations to be built, so this was my attempt to try and push that responsibility back into the Headers module. I'm willing to pull that back into Response though, if you think it makes more sense for it to be there!

Reply:
I could also build specific functions for each header type, like `build_allowed_methods_headers` or `build_redirect_headers`. So that way the conditional statements that determine which headers to use would still exist in `build/4`, but it's not building the headers itself.

Reply:
Just fixed this so instead of passing the whole response object, Headers.build takes request, status_code, body and media_type!